### PR TITLE
`MessageField` and `ThrowableMessage` improvements

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR
  * Introduces `MessageFieldException` is introduced, which improves diagnostics for error cases.
  * Adds the ability to set producer ID for a `ThrowableMessage`.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, PTAL.

File: base/src/main/java/io/spine/protobuf/MessageFieldException.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package io.spine.protobuf;
+
+import com.google.protobuf.Message;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Signals an error working with a protobuf message field.
```
Comment (armiol) [inline base/src/main/java/io/spine/protobuf/MessageFieldException.java]: Let's call it `Protobuf`, with a capital.

File: base/src/main/java/io/spine/protobuf/MessageFieldException.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    /**
+     * Obtains a Protobuf message working with a field of which caused an error
+     */
+    public Message getProtobufMessage() {
+        return protobufMessage;
+    }
+
+    private static String formatMessage(String format, @Nullable Object... params) {
+        if (params == null) {
```
Comment (armiol) [inline base/src/main/java/io/spine/protobuf/MessageFieldException.java]: This method is reported to be only partially tested.

File: base/src/main/java/io/spine/protobuf/MessageFieldException.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+public class MessageFieldException extends RuntimeException {
+
+    private final Message protobufMessage;
+
+    /**
+     * Constructs a {@code MessageFieldException} with the formatted message text.
+     *
+     * @param protobufMessage    a Protobuf message object working with a field of which caused
+     *                           an error
+     * @param errorMessageFormat a formatted string for error message
```
Comment (armiol) [inline base/src/main/java/io/spine/protobuf/MessageFieldException.java]: I would not call that a `formatted string`. That's a string template used for error message formatting.

**Pull request merged**